src/Overlay.js

```diff
@@ -42,56 +35,63 @@ class Overlay extends React.Component {
       , animation: Transition
       , ...props } = this.props;
 
-    let child = null;
-
-    if ( Transition === true ){
+    if (Transition === true) {
       Transition = Fade;
     }
 
-    if (props.show || (Transition && this.state.exiting && !this.state.exited)) {
+    // Don't un-render the overlay while it's transitioning out.
+    const mountOverlay = props.show || (Transition && !this.state.exited);
```

Comment:
I think there were a few edge cases around the "exited" trigger firing early from `onExited` - since I now explicitly cancel the chained callbacks in `Transition`, those no longer come up.